### PR TITLE
feat(anthropic): add Anthropic component

### PR DESCRIPTION
**Pull request by namwoam**
Summary

1. Added support for the Anthropic component with fields aligned to the current Instill model's "TASK_TEXT_GENERATION_CHAT" task.
2. Added "GetBase64FileExtension" helper function and tests to the base component.

### CODE REVIEW

**Review by chuang8511 — DISMISSED** [review]
LGTM.
- [previous conversation](https://github.com/instill-ai/component/pull/161#pullrequestreview-2135628075)
Will need HuiTang to review it as a final review.

File: ai/anthropic/v0/main.go
Comment (donch1989) [inline ai/anthropic/v0/main.go:163]: WaitGroup is not a lock.

File: ai/anthropic/v0/main.go
Comment (donch1989) [inline ai/anthropic/v0/main.go:156]: There is a high possibility of a thread-safety issue here. We should use a lock.
However, in this case, we don’t even need a goroutine since these operations are all CPU-bound. There’s no benefit to using a goroutine here.

File: ai/anthropic/v0/main.go
Comment (donch1989) [inline ai/anthropic/v0/main.go:187]:
```suggestion
	chatHistory := in.Fields["chat-history"].GetListValue()
```

File: base/formats.go
Comment (donch1989) [inline base/formats.go:203]: Did we use this function somewhere? If not, let's remove it and rename `GetBase64FileExtensionFast` to `GetBase64FileExtension`

**Review by donch1989 — CHANGES_REQUESTED** [review]

**Comment by donch1989** [comment]
Hi @namwoam @chuang8511 ,
Thanks for your work. I've added some comments. Please take a look.

File: ai/anthropic/v0/main.go
Comment (namwoam) [reply ai/anthropic/v0/main.go]: fixed

File: ai/anthropic/v0/main.go
Comment (namwoam) [reply ai/anthropic/v0/main.go]: Ok, changed the implementation to 2 layer for loop.

File: base/formats.go
Comment (namwoam) [reply base/formats.go]: Fixed.

**Review by namwoam — COMMENTED** [review]
Ok, changed the implementation to 2 layer for loop.

File: ai/anthropic/v0/config/setup.json
Comment (donch1989) [inline ai/anthropic/v0/config/setup.json:13]:
```suggestion
        "instillSecret": true,
```

**Review by donch1989 — CHANGES_REQUESTED** [review]
After this is fixed, the PR can be merged.

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**
